R/snapshot.R

```diff
@@ -741,54 +741,38 @@ renv_snapshot_description_impl <- function(dcf, path = NULL) {
     dcf <- dcf[fields]
   }
 
+  # drop fields that normally only appear in binary packages,
+  # or fields which might differ from user to user, or might
+  # differ depending on the mirror used for publication
+  ignore <- c("Archs", "Built", "Date/Publication", "File", "MD5sum", "Packaged")
+  dcf[ignore] <- NULL
+  
+  # drop remote fields for cranlike remotes
+  if (renv_record_cranlike(dcf)) {
+    sha <- dcf[["RemoteSha"]]
+    if (is.null(sha) || nchar(sha) < 40L) {
+      remotes <- grep("^Remote", names(dcf), perl = TRUE, value = TRUE)
+      dcf[remotes] <- NULL
+    }
+  }
+  
+  # drop the old Github remote fields
+  github <- grepl("^Github", names(dcf), perl = TRUE)
```

Comment:
aside: When `RemoteType: github` and `RemoteHost: api.github.com`, each of the remote `Repo`, `Username`, `Ref`, and `Sha` fields can be transformed into the `Github` equivalents (`Sha` becoming `SHA1`).